src/components/organisms/batch-jobs-activity-list/index.tsx

```diff
@@ -0,0 +1,171 @@
+import useNotification from "../../../hooks/use-notification"
+import { useAdminCancelBatchJob } from "medusa-react"
+import React, { useEffect } from "react"
+import { getErrorMessage } from "../../../utils/error-messages"
+import getRelativeTime from "../../../utils/get-human-relative-time"
+import { getActivityDescriptionFromBatchJob } from "./utils"
+import MedusaIcon from "../../fundamentals/icons/medusa-icon"
+import Button from "../../fundamentals/button"
+import DownloadableFileButton from "../../molecules/downloadable-file-button"
+import Medusa from "../../../services/api"
+import { BatchJob } from "@medusajs/medusa/dist"
+import { bytesConverter } from "../../../utils/bytes-converter"
+import { ActivityCard } from "../../molecules/activity-card"
+
+const BatchJobActivityList = ({ batchJobs }: { batchJobs?: BatchJob[] }) => {
+  return <div>
+    {
+      !!batchJobs?.length && (
+        batchJobs?.map(batchJob => {
+          return <BatchJobActivityCard key={batchJob.id} batchJob={batchJob}/>
+        })
+      )
+    }
+  </div>
+}
+
+const BatchJobActivityCard = ({ batchJob }: { batchJob: any }) => {
+  const notification = useNotification()
+  const { mutate: cancelBatchJob, error: cancelBatchJobError  } =
+    useAdminCancelBatchJob(batchJob.id)
+
+  const relativeTimeElapsed = getRelativeTime({
+    from: new Date(),
+    to: batchJob.created_at,
+  })
+
+  const canCancel = !batchJob.completed_at
+    && !batchJob.failed_at
+    && !batchJob.canceled_at
+
+  const canDownload = !!batchJob.completed_at
+    && !batchJob.failed_at
+    && !batchJob.canceled_at
+    && batchJob.result?.file_key
```

Comment:
would status checks not be enough here? (just for clarity if we have the enum and versioning between packages work)